File: wbj-main/src/main/java/com/app/pojos/Mechanic.java
package com.app.pojos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "mechanic")
public class Mechanic extends BaseEntity {
	
	public Mechanic() {
		
	}
	
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JoinColumn(name = "mechanic_id")
    private Long id;

    @Column(length = 255, nullable = false)
    @JoinColumn(name = "user_name")
    private String name;

    @Column(length = 255, nullable = false, unique = true)
    private String email;

    @Column(length = 255, nullable = false)
    private String password;

    @Column(length = 15, nullable = false)
    private String contactno;

    @Column(length = 15, nullable = false)
    private Long age;

    @Temporal(TemporalType.TIMESTAMP)
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "create_date")
    private Date createDate;

    @Column(length = 255, nullable = false)
    private String location;

    @Column(length = 255, nullable = false)
    private String address;

    @Column(length = 255, nullable = false)
    private String gender;
    
    

    public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getContactno() {
		return contactno;
	}

	public void setContactno(String contactno) {
		this.contactno = contactno;
	}

	public Long getAge() {
		return age;
	}

	public void setAge(Long age) {
		this.age = age;
	}

	public Date getCreateDate() {
		return createDate;
	}

	public void setCreateDate(Date createDate) {
		this.createDate = createDate;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public String getGender() {
		return gender;
	}

	public void setGender(String gender) {
		this.gender = gender;
	}

	@Override
    public String toString() {
        return "Mechanic{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", password='" + password + '\'' +
                ", contactno='" + contactno + '\'' +
                ", age=" + age +
                ", createDate=" + createDate +
                ", location='" + location + '\'' +
                ", address='" + address + '\'' +
                ", gender='" + gender + '\'' +
                '}';
    }

	public Mechanic(Long id, Long id2, String name, String email, String password, String contactno, Long age,
			Date createDate, String location, String address, String gender) {
		super(id);
		id = id2;
		this.name = name;
		this.email = email;
		this.password = password;
		this.contactno = contactno;
		this.age = age;
		this.createDate = createDate;
		this.location = location;
		this.address = address;
		this.gender = gender;
	}
}

File: dotnetbackend/Model/Mechanic.cs
namespace dotnetbackend.Model;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("mechanic")]
public class Mechanic
    {
       [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("mechanic_id")]
        public long Id { get; set; }

        [Required]
        [StringLength(255)]
        [Column("mechanic_name")]
        public string Name { get; set; }

        [Required]
        [StringLength(255)]
        public string Email { get; set; }

        [Required]
        [StringLength(255)]
        public string Password { get; set; }

        [Required]
        [StringLength(15)]
        public string Contactno { get; set; }

        [Required]
        [StringLength(15)]
        public long Age { get; set; }

        [Required]
        [Column("created_at", TypeName = "datetime2")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime CreateDate { get; set; }

        [Required]
        [StringLength(255)]
        public string Location { get; set; }

        [Required]
        [StringLength(255)]
        public string Address { get; set; }

        [Required]
        [StringLength(255)]
        public string Gender { get; set; }


        public Mechanic(long id, string name, string email, string password, string contactno, long age, DateTime createDate, string address, string location, string gender)
        {
            Id = id;
            Name = name;
            Email = email;
            Password = password;
            Contactno = contactno;
            Age = age;
            CreateDate = createDate;
            Address = address;
            Location = location;
            Gender = gender;
        }

        public override string ToString()
        {
            return $"Mechanic {{ Id={Id}, Name={Name}, Email={Email}, Password={Password}, ContactNo={Contactno}, Age={Age}, CreateDate={CreateDate}, Address={Address}, Location={Location}, Gender={Gender} }}";
        }
    }
